### PR TITLE
(chore) - Fix react-static docs pages build

### PR DESCRIPTION
**Pull request by kitten**
- Fix react-router plugin not outputting correct base path for static export
- Replace relative routes with absolute ones for MDX output (md-plugin@0.3.0)
- Replace sidebar links with absolute ones
- Fix basepath for Google Analytics component as per Victory docs
- Fix basepath across site
- Downgrade react-static to fix static export (otherwise HTML output is empty)

This took a while to figure out as the behaviour and exports is just very
messed up. The downgrade is needed, otherwise no static HTML will be exported.

### CODE REVIEW

File: packages/site/static.config.js [hunk 1]
```diff
@@ -4,12 +4,12 @@ import constants from './src/constants';
...
 
 const isStaging = process.env.REACT_STATIC_STAGING === 'true';
-const basePath = 'open-source/urql';
 
 export default {
   plugins: [
     resolve(__dirname, 'plugins/monorepo-fix/'),
-    resolve(__dirname, 'plugins/preact'),
+    resolve(__dirname, 'plugins/react-router/'),
+    // resolve(__dirname, 'plugins/preact/'),
```
Comment (wgolledge) [inline packages/site/static.config.js:12]: uncomment?

File: packages/site/plugins/react-router/browser.api.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+const Location = withRouter(({ children, location }) => children(location))
+
+export default ({ RouterProps: userRouterProps = {} }) => ({
+  Root: PreviousRoot => ({ children }) => {
+    const routerProps = { basename: useBasepath() || '' }
+    if (routerProps.basename) routerProps.basename = `/${routerProps.basename}`;
+    const staticInfo = useStaticInfo()
+
+    // Test for document to detect the node stage
+    let Router
```
Comment (wgolledge) [inline packages/site/plugins/react-router/browser.api.js]: Semi-colons a little bit inconsistent?

File: packages/site/plugins/react-router/browser.api.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const routerProps = { basename: useBasepath() || '' };
+    if (routerProps.basename) routerProps.basename = `/${routerProps.basename}`;
+    const staticInfo = useStaticInfo();
+
+    // Test for document to detect the node stage
+    let Router;
+    if (typeof document !== 'undefined') {
+      // NOTE: React Router is inconsistent in how it handles base paths
+      // This will need a trailing slash while the StaticRouter does not
+      if (routerProps.basename) routerProps.basename += '/';
```
Comment (kitten) [inline packages/site/plugins/react-router/browser.api.js:21]: if you want to see something wild, look no more 😆

**Review by wgolledge — APPROVED** [review]
This is working great as far as I can tell & I'm a big fan of the absolute links!

It's a shame about the downgrade (especially going forward) but it seems to be pretty necessary

File: packages/site/plugins/react-router/browser.api.js (hunk 3 above)
Comment (wgolledge) [reply packages/site/plugins/react-router/browser.api.js:21]: Yea this is bonkers

File: packages/site/static.config.js (hunk 1 above)
Comment (kitten) [reply packages/site/static.config.js:12]: oh, I haven't tested this yet; give me a second

File: packages/site/plugins/react-router/browser.api.js (hunk 2 above)
Comment (kitten) [reply packages/site/plugins/react-router/browser.api.js]: that's because it's a vendored file; I'll make it more consistent

**Pull request merged**